requirements.txt

```diff
@@ -18,3 +18,4 @@ tqdm>=4.41.0
 PyYAML>=5.3.1
 wget
 requests_toolbelt
+ipython>=8.9.0
```

Comment:
How heavy is this additional requirement? 

We probably should put some effort into really slimming down our list of requirements so installing our pip package isn’t such a burden (OpenCV and matplotlib in particular are beasts).

Reply:
we can have additional installs in `setup.py`, e.g. for `notebook` is setup. Once loaded, the package can ask the runtime where he is and download additional stuff 

I think opencv is needed in general, not sure why we have `matplotlib` is we have opencv/pillow